Index: plymouth/src/libply-splash-core/ply-keyboard.c
===================================================================
--- plymouth.orig/src/libply-splash-core/ply-keyboard.c
+++ plymouth/src/libply-splash-core/ply-keyboard.c
@@ -262,6 +262,7 @@ on_key_event (ply_keyboard_t *keyboard,
 {
         const char *bytes;
         size_t size, i;
+        bool has_escape = false;
 
         bytes = ply_buffer_get_bytes (buffer);
         size = ply_buffer_get_size (buffer);
@@ -284,8 +285,15 @@ on_key_event (ply_keyboard_t *keyboard,
                 }
 
                 keyboard_input = strndup (bytes + i, character_size);
+      
+                wchar_t key;
+                if ((ssize_t) mbrtowc (&key, keyboard_input, character_size, NULL) > 0) {
+                  if (key == KEY_ESCAPE)
+                    has_escape = true;
+                }
 
-                process_keyboard_input (keyboard, keyboard_input, character_size);
+                if (has_escape == false)
+                  process_keyboard_input (keyboard, keyboard_input, character_size);
 
                 i += character_size;
 
